src/core/experience/service/experience.service.ts

```diff
@@ -72,6 +72,22 @@ export class ExperienceService {
     return;
   }
 
+  async removeMyExperience(
+    id: number,
+    user: UserActiveInterface,
+  ): Promise<void> {
+    const experienceToRemove = await this.experienceRepository.softDelete({
+      profileId: user.id,
+      id: id,
+    });
+
+    if (experienceToRemove.affected === 0) {
+      throw new NotFoundException(EXPERIENCE_NOT_FOUND);
+    }
```

Comment:
Pienso que lo correcto es hacer una pre-validacion e.e
```suggestion
    const experience = await this.experienceRepository.findOne({ id });
    if (!experience) throw new NotFoundException(EXPERIENCE_NOT_FOUND);
    
    try {
        const experienceToRemove = await this.experienceRepository.softDelete({
           profileId: user.id,
           id: id,
        });

        if (experienceToRemove.affected === 0) {
           throw new InternalServerErrorException(UNKOWN_ERROR_DATABASE_MESSAGE);
        }
    } catch (error) {
       if (error instanceof InternalServerErrorException) throw error;
       throw new InternalServerErrorException(UNKOWN_ERROR_DATABASE_MESSAGE);
    }

```

Reply:
Ademas, tambien nos permite hacer una validacion mas especializada
```suggestion
    const experience = await this.experienceRepository.findOne({ id });
    if (!experience) throw new NotFoundException(EXPERIENCE_NOT_FOUND);
    if (experience.profileId !== user.id) throw new ForbiddenException(YOU_NOT_HAVE_PERMISSION_MESSAGE);
    
    try {
        const experienceToRemove = await this.experienceRepository.softDelete({
           profileId: user.id,
           id: id,
        });
        if (experienceToRemove.affected === 0) {
           throw new InternalServerErrorException(UNKOWN_ERROR_DATABASE_MESSAGE);
        }
    } catch (error) {
       if (error instanceof InternalServerErrorException) throw error;
       throw new InternalServerErrorException(UNKOWN_ERROR_DATABASE_MESSAGE);
    }
```
PD: notese que uso `ForbiddenException` en lugar de `UnauthorizedException` ya que Unauthorized se usa cuando el usuario no esta autenticado ni tiene token, Forbidden es para cuando si bien esta autenticado, no tiene acceso a cierto recurso o cierto permiso

Reply:
Me gustaria hicieran esto en este caso en concreto para tener algo de logica de negocio con prevalidaciones guardado por alli si llegamos a necesitarlo para un caso donde si o si necesitemos hacerlo de esta forma

Para el resto de casos donde simplemente se limitan a hacer `experienceToRemove.affected === 0` la verdad no lo se... diria que son deuda tecnica, pero no se si a partir de ahora sea bueno complicarse la vida haciendo esto o irrnos por lo simple (que no estoy seguro funcione en el 100% de los casos)

Queda de parte de la investigacion del manejo de errores en TypeORM que no se quien la esta haciendo